[🐛 Bug]:  [java] No proper error message misleading the user to analyze

### Description


**Error : Session is not created : Possible cause invalid address of the remote server or browser start up failed**
Please refer the appium java client issue and inturn it uses selenium  client
[https://github.com/appium/java-client/issues/2349](url) 

[  The problem is here.
when we are using Desired capabilities as option on Appium 3.0 server, it throws generic error session not created
as we know - Desired capabilities option is depreacted on 3.0
Expected : Its should throw Deprecated or unsupported desired capabilities or compile time error ]
------------------------------------------------------------------------------------------------


JAVA Version - 11/17/23

**Appium server - 3.0.2
Appium client - 10.0.0
**Selenium-java - 4.35.0**
IOS - XCUI test driver - 10.1.2
Testng- 7.5.1
Server URL : http://127.0.0.1:4723/
Even cross check with compatible matrix appium client & Selenium , Didnt helped me**


### Reproducible Code

```shell
// This sample code supports Appium Java client >=9
import org.testng.annotations.Test;
import io.appium.java_client.ios.IOSDriver;
import io.appium.java_client.ios.options.XCUITestOptions;

import java.net.MalformedURLException;
import java.net.URL;

import org.openqa.selenium.remote.DesiredCapabilities;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
public class NewTest2 {

	IOSDriver driver; 

	@BeforeClass
	public void setUp() {
		DesiredCapabilities capabilities = new DesiredCapabilities();
		capabilities.setCapability("platformName", "iOS");
		capabilities.setCapability("acceptAlerts", true);
		capabilities.setCapability("trust", true);
		capabilities.setCapability("launchTimeout", 600000);
		capabilities.setCapability("newCommandTimeout", 600);
		capabilities.setCapability("automationName", "XCUITest");
	    capabilities.setCapability("app", "/Users/ga20055252/Downloads/uat-Runner.app");
		capabilities.setCapability("udid", "CB53FD24-EE7B-4EA8-BE4A-C3D88FA1AC1A");
		capabilities.setCapability("noReset", true);
		capabilities.setCapability("bundleId", "com.mgmresorts.enterprise.uat");
		capabilities.setCapability("wdaLaunchTimeout", 60000);
		capabilities.setCapability("autoDismissAlerts", true);
		capabilities.setCapability("fullReset", false);

		try {
			URL appiumServerUrl = new URL("http://127.0.0.1:4723/wd/hub");
			driver = new IOSDriver(appiumServerUrl, capabilities);
		} catch (MalformedURLException e) {	e.printStackTrace();}}

	@Test
	public void sampleTest() {
		try {Thread.sleep(3000);} catch (InterruptedException e) {e.printStackTrace();}
	}

	@AfterClass
	public void tearDown() {
		driver.quit();
	}
}
```

### Debugging Logs

```logs
Session is not created | No logs at all
```

```
org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure. 
Host info: host: 'C02HF6S6Q6L5', ip: '2401:4900:8f4e:f78c:c936:5fdc:fb7b:90e6%en0'
Build info: version: '4.36.0', revision: '6d115cfd87'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '15.6.1', java.version: '11.0.28'
Driver info: io.appium.java_client.ios.IOSDriver
Command: [null, newSession {capabilities=[Capabilities {acceptAlerts: true, app: /Users/ga20055252/Downloads..., autoDismissAlerts: true, automationName: XCUITest, bundleId: com.mgmresorts.enterprise.uat, fullReset: false, launchTimeout: 600000, newCommandTimeout: 600, noReset: true, platformName: IOS, trust: true, udid: CB53FD24-EE7B-4EA8-BE4A-C3D..., wdaLaunchTimeout: 60000}]}]
Capabilities {acceptAlerts: true, app: /Users/ga20055252/Downloads..., autoDismissAlerts: true, automationName: XCUITest, bundleId: com.mgmresorts.enterprise.uat, fullReset: false, launchTimeout: 600000, newCommandTimeout: 600, noReset: true, platformName: IOS, trust: true, udid: CB53FD24-EE7B-4EA8-BE4A-C3D..., wdaLaunchTimeout: 60000}
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:575)
	at io.appium.java_client.AppiumDriver.startSession(AppiumDriver.java:308)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:174)
	at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:98)
	at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:109)
	at io.appium.java_client.ios.IOSDriver.<init>(IOSDriver.java:100)
	at NewTest2.setUp(NewTest2.java:35)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:136)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:65)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:381)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:319)
	at org.testng.internal.invokers.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:178)
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:122)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.testng.TestRunner.privateRun(TestRunner.java:808)
	at org.testng.TestRunner.run(TestRunner.java:603)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
	at org.testng.SuiteRunner.run(SuiteRunner.java:326)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.runSuites(TestNG.java:1092)
	at org.testng.TestNG.run(TestNG.java:1060)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
Caused by: java.lang.IllegalArgumentException: Illegal key values seen in w3c capabilities: [acceptAlerts, app, autoDismissAlerts, automationName, bundleId, fullReset, launchTimeout, newCommandTimeout, noReset, trust, udid, wdaLaunchTimeout]
	at org.openqa.selenium.remote.NewSessionPayload.lambda$validate$5(NewSessionPayload.java:163)
	at java.base/java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:441)
	at java.base/java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:442)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:442)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.openqa.selenium.remote.NewSessionPayload.validate(NewSessionPayload.java:167)
	at org.openqa.selenium.remote.NewSessionPayload.<init>(NewSessionPayload.java:70)
	at org.openqa.selenium.remote.NewSessionPayload.create(NewSessionPayload.java:99)
	at org.openqa.selenium.remote.NewSessionPayload.create(NewSessionPayload.java:84)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:60)
	at io.appium.java_client.remote.AppiumCommandExecutor.createSession(AppiumCommandExecutor.java:178)
	at io.appium.java_client.remote.AppiumCommandExecutor.execute(AppiumCommandExecutor.java:243)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:557)
	... 32 more
```

--- a/java/src/org/openqa/selenium/remote/RemoteWebDriver.java
+++ b/java/src/org/openqa/selenium/remote/RemoteWebDriver.java
@@ -572,10 +572,14 @@ public class RemoteWebDriver
         if (e instanceof SessionNotCreatedException) {
           toThrow = (WebDriverException) e;
         } else {
+          // When this exception comes from a remote end, the real cause is usually hidden in the
+          // cause. Let's try to rescue it and display it at the top level.
+          String cause = e.getCause() != null ? e.getCause().getMessage() : "";
           toThrow =
               new SessionNotCreatedException(
                   "Possible causes are invalid address of the remote server or browser start-up"
-                      + " failure.",
+                      + " failure. "
+                      + cause,
                   e);
         }
       } else if (e instanceof WebDriverException) {

--- a/java/src/org/openqa/selenium/remote/RemoteWebDriver.java
+++ b/java/src/org/openqa/selenium/remote/RemoteWebDriver.java
@@ -574,11 +574,11 @@ public class RemoteWebDriver
         } else {
           // When this exception comes from a remote end, the real cause is usually hidden in the
           // cause. Let's try to rescue it and display it at the top level.
-          String cause = e.getCause() != null ? e.getCause().getMessage() : "";
+          String cause = e.getCause() != null ? " " + e.getCause().getMessage() : "";
           toThrow =
               new SessionNotCreatedException(
                   "Possible causes are invalid address of the remote server or browser start-up"
-                      + " failure. "
+                      + " failure."
                       + cause,
                   e);
         }

